recreate.go

```diff
@@ -0,0 +1,442 @@
+				return true
+			}
+		}
+		return false
+	})
+	return sortedTypes
+}
+
+func (km *KeyspaceMetadata) tableToCQL(w io.Writer, kn string, tm *TableMetadata) error {
+	t := template.Must(template.New("table").
+		Funcs(map[string]interface{}{
+			"escape":               km.escape,
+			"tableColumnToCQL":     km.tableColumnToCQL,
+			"tablePropertiesToCQL": km.tablePropertiesToCQL,
+		}).
+		Parse(`
+CREATE TABLE {{ .KeyspaceName }}.{{ .Tm.Name }} (
+  {{ tableColumnToCQL .Tm }}
+) WITH {{ tablePropertiesToCQL .Tm.ClusteringColumns .Tm.Options .Tm.Flags }};
+`))
+	if err := t.Execute(w, map[string]interface{}{
+		"Tm":           tm,
+		"KeyspaceName": kn,
+	}); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (km *KeyspaceMetadata) tableColumnToCQL(tm *TableMetadata) string {
+	var sb strings.Builder
+
+	var columns []string
+	for _, cn := range tm.OrderedColumns {
+		cm := tm.Columns[cn]
+		column := fmt.Sprintf("%s %s", cn, cm.Type)
+		if cm.Kind == ColumnStatic {
+			column += " static"
+		}
+		columns = append(columns, column)
+	}
+	if len(tm.PartitionKey) == 1 && len(tm.ClusteringColumns) == 0 && len(columns) > 0 {
+		columns[0] += " PRIMARY KEY"
+	}
+
+	sb.WriteString(strings.Join(columns, ",\n  "))
+
+	if len(tm.PartitionKey) > 1 || len(tm.ClusteringColumns) > 0 {
+		sb.WriteString(",\n  PRIMARY KEY (")
+		sb.WriteString(km.partitionKeyString(tm.PartitionKey, tm.ClusteringColumns))
+		sb.WriteRune(')')
+	}
+
+	return sb.String()
+}
+
+func (km *KeyspaceMetadata) partitionKeyString(pks, cks []*ColumnMetadata) string {
+	var sb strings.Builder
+
+	if len(pks) > 1 {
+		sb.WriteRune('(')
+		for i, pk := range pks {
+			if i != 0 {
+				sb.WriteString(", ")
+			}
+			sb.WriteString(pk.Name)
+		}
+		sb.WriteRune(')')
+	} else {
+		sb.WriteString(pks[0].Name)
+	}
+
+	if len(cks) > 0 {
+		sb.WriteString(", ")
+		for i, ck := range cks {
+			if i != 0 {
+				sb.WriteString(", ")
+			}
+			sb.WriteString(ck.Name)
+		}
+	}
+
+	return sb.String()
+}
+
+func (km *KeyspaceMetadata) tablePropertiesToCQL(cks []*ColumnMetadata, opts TableMetadataOptions, flags []string) (string, error) {
+	var sb strings.Builder
+
+	var properties []string
+
+	compactStorage := len(flags) > 0 && (contains(flags, TableFlagDense) ||
+		contains(flags, TableFlagSuper) ||
+		!contains(flags, TableFlagCompound))
+
+	if compactStorage {
+		properties = append(properties, "COMPACT STORAGE")
+	}
+
+	if len(cks) > 0 {
+		var inner []string
+		for _, col := range cks {
+			inner = append(inner, fmt.Sprintf("%s %s", col.Name, col.ClusteringOrder))
+		}
+		properties = append(properties, fmt.Sprintf("CLUSTERING ORDER BY (%s)", strings.Join(inner, ", ")))
+	}
+
+	options, err := km.tableOptionsToCQL(opts)
+	if err != nil {
+		return "", err
+	}
+	properties = append(properties, options...)
+
+	sb.WriteString(strings.Join(properties, "\n    AND "))
+	return sb.String(), nil
+}
+
+func (km *KeyspaceMetadata) tableOptionsToCQL(ops TableMetadataOptions) ([]string, error) {
```

Comment:
@zimnx this list is not all, for instance where is [scylla_encryption_options](https://docs.scylladb.com/operating-scylla/security/encryption-at-rest/)?

Reply:
I have checked and it looks like in memory and encryption are the ony ones

```
 michal@lh:scylla/scylla ((scylla-2019.1.7))$ rg '(\.|->)add_[a-z_]*extension'
mirror-file-impl.cc
281:            exts.add_sstable_file_io_extension("in_memory_file_store", std::make_unique<in_memory_file_ext>());

tests/extensions_test.cc
54:    ext->add_schema_extension("los_lobos", [](db::extensions::schema_ext_config cfg) {
91:    ext->add_sstable_file_io_extension("los_lobos", std::make_unique<dummy_ext>());

ent/encryption/encryption.cc
583:    exts.add_schema_extension(encryption_attribute, [ctxt](auto v) {
586:    exts.add_sstable_file_io_extension(encryption_attribute, std::make_unique<encryption_file_io_extension>(ctxt));
604:        exts.add_commitlog_file_extension(encryption_attribute, std::make_unique<encryption_commitlog_file_extension>(ctxt, opts));
615:                    exts.add_extension_to_schema(s, encryption_attribute, f(opts));
```